test/linalg.jl

```diff
@@ -31,38 +38,45 @@ using DataFrames, LinearAlgebra, MixedModels, Random, SparseArrays, StatsModels,
-        G = repeat('A':'T', inner = 2, outer=10), H = repeat('a':'j', inner=40))
-    lmm1 = fit!(LinearMixedModel(@formula(Y ~ 1+A+(1+A|G)+(1+A|H)), df,
-        wts  = ones(400)))
-    @test loglikelihood(lmm1) ≈ -578.9080978272708
-    MixedModels.reweight!(lmm1, ones(400))
-    @test loglikelihood(fit!(lmm1)) ≈ -578.9080978272708
+    rng = MersenneTwister(1234321)
+    df = (
+        Y = randn(rng, 400),
+        A = repeat(PooledArray(["N","Y"]), outer=200),
+        G = repeat(PooledArray(string.('A':'T')), inner = 2, outer=10),
+        H = repeat(PooledArray(string.('a':'j')), inner=40),
+        )
+    m1 = fit(MixedModel, @formula(Y ~ 1 + A + (1+A|G) + (1+A|H)), df)
+    wm1 = fit(MixedModel, @formula(Y ~ 1+A+(1+A|G)+(1+A|H)), df, wts  = ones(400))
+    @test loglikelihood(wm1) ≈ loglikelihood(m1)
+    MixedModels.reweight!(wm1, ones(400))
+    @test loglikelihood(fit!(wm1)) ≈ loglikelihood(m1)
 end
 
 @testset "rankupdate!" begin
-    @test ones(2, 2) == MixedModels.rankUpdate!(Hermitian(zeros(2, 2)), ones(2))
+    @test ones(2, 2) == rankUpdate!(Hermitian(zeros(2, 2)), ones(2))
     d2 = Diagonal(fill(2., 2))
-    @test Diagonal(fill(5.,2)) == MixedModels.rankUpdate!(Diagonal(ones(2)), d2, 1.)
-    @test Diagonal(fill(-3.,2)) == MixedModels.rankUpdate!(Diagonal(ones(2)), d2, -1.)
+    @test Diagonal(fill(5.,2)) == rankUpdate!(Diagonal(ones(2)), d2, 1.)
+    @test Diagonal(fill(-3.,2)) == rankUpdate!(Diagonal(ones(2)), d2, -1.)
 
     # when converting straight from diagonal to symmetric, the type is different
-    @test Diagonal(fill(5.,2)) == MixedModels.rankUpdate!(Symmetric(Matrix(1. * I(2)), :L), d2)
+    @test Diagonal(fill(5.,2)) == rankUpdate!(Symmetric(Matrix(1. * I(2)), :L), d2)
     # generic method
-    @test Diagonal(fill(5.,2)) == MixedModels.rankUpdate!(Matrix(1. * I(2)), d2)
+    @test Diagonal(fill(5.,2)) == rankUpdate!(Matrix(1. * I(2)), d2)
 end
 
 @testset "lmulλ!" begin
-    gendata(n::Int, ng::Int) = gendata(MersenneTwister(42), n, ng)
+    levs(ng, tag='S') = string.(tag, lpad.(string.(1:ng), ndigits(ng), '0'))
 
-    function gendata(rng::AbstractRNG, n::Int, ng::Int)
-        df = DataFrame(Y = randn(rng, n),
-                       X = rand(rng, n),
-                       G = rand(rng, 1:ng, n),
-                       H = rand(rng, 1:ng, n))
-        categorical!(df, [:G, :H])
+    function gendata(rng::AbstractRNG, n::Integer, ng::Integer, nh::Integer)
+        (
+            Y = randn(rng, n),
+            X = rand(rng, n),
+            G = PooledArray(rand(rng, levs(ng, 'G'), n)),
+            H = PooledArray(rand(rng, levs(nh, 'H'), n)),
+        )
     end
+    gendata(n::Integer, ng::Integer, nh::Integer) = gendata(MersenneTwister(42), n, ng, nh)
+    gendata(n::Integer, ng::Integer) = gendata(MersenneTwister(42), n, ng, ng)
 
     @testset "Adjoint{T,ReMat{T,1}}, BlockedSparse{T,1,2}" begin
         # this is an indirect test of lmulΛ! for a blocking structure found in
```

Comment:
I'm not sure this test is still doing what we want it to do for other random numbers. From the [original issue](https://github.com/JuliaStats/MixedModels.jl/issues/123#issuecomment-535497527 ):
> This highlights the point about needing a reproducible RNG -- the particular blocking here arises from a particular set of pseudorandom numbers.